File: 78_Sine_Wave/ruby/sinewave.rb
def intro
  puts "                              SINE WAVE
               CREATIVE COMPUTING  MORRISTOWN, NEW JERSEY\n\n\n\n\n\n"
end

def main
  intro
  (0..40).step(0.25).each do |t|
    a = (26 + 25 * Math.sin(t)).to_i
    text = (t % 0.5) == 0 ? "CREATIVE" : "COMPUTING"
    puts " " * a + text
  end
end

main

File: 78_Sine_Wave/javascript/sinewave.mjs
#!/usr/bin/env node

import { println, tab } from '../../00_Common/javascript/common.mjs';

println(tab(30), "SINE WAVE");
println(tab(15), "CREATIVE COMPUTING  MORRISTOWN, NEW JERSEY");
println("\n".repeat(4));

// REMARKABLE PROGRAM BY DAVID AHL
// Transliterated to Javascript by Les Orchard <me@lmorchard.com>

let toggleWord = true;

for (let step = 0; step < 40; step += 0.25) {
  let indent = Math.floor(26 + 25 * Math.sin(step));
  println(tab(indent), toggleWord ? "CREATIVE" : "COMPUTING");
  toggleWord = !toggleWord;
}
